controllers/activemqartemis_reconciler.go

```diff
@@ -1989,12 +1979,44 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
 	return pts, nil
 }
 
-func brokerPropertiesConfigSystemPropValue(mountPoint, resourceName string, brokerPropertiesData map[string]string) string {
+func configureBrokerPropertiesMount(podSpec *corev1.PodSpec, resourceName string, isSecret bool, brokerPropertiesMapData map[string]string) {
+
+	// this depends on init container passing --java-opts to artemis create via launch.sh *and* it
+	// not getting munged on the way. We CreateOrAppend to any value from spec.Env
+	var mountPoint = secretPathBase
+	if !isSecret {
+		mountPoint = cfgMapPathBase
+	}
+
+	javaOpts := corev1.EnvVar{
+		Name:  "JAVA_OPTS",
+		Value: brokerPropertiesConfigFileSystemPropValue(mountPoint, resourceName, brokerPropertiesMapData),
+	}
+	environments.CreateOrAppend(podSpec.InitContainers, &javaOpts)
+
+	// older versions won't use with JAVA_ARGS_APPEND and these will prevale over JAVA_OPTS
+	javaArgsAppend := corev1.EnvVar{
+		Name:  "JAVA_ARGS_APPEND",
+		Value: brokerPropertiesConfigFolderSystemPropValue(mountPoint, resourceName, brokerPropertiesMapData),
+	}
+	environments.CreateOrAppend(podSpec.Containers, &javaArgsAppend)
```

Comment:
Why do we need `brokerPropertiesConfigFolderSystemPropValue`? Does `brokerPropertiesConfigFileSystemPropValue` work with older and newer versions?

Reply:
2.27.1 introduced support for treating the brokerProperties url ending in a / as a directory. That version also introduced JAVA_ARGS_APPEND.
So the 'folder' via the new env var is what we want, it gives us freedom to ignore the name or the properties file. But for existing deployments that just mount a single immutable properties file, we need to respect the use of the file url.

Reply:
yes, brokerPropertiesConfigFileSystemPropValue works with old and new.

Reply:
I see, I was wondering that a single implementati on could be more easy to maintain if both implementations are equivalent, anyway I have no strong opinioni on this, It LGTM thanks for this fix